File: 4_single_inheritance.cpp


#include <bits/stdc++.h>
using namespace std;
#define ll long long

class Animal{
    public:
    string name;
    Animal(string name){
        this->name = name;
    }
    void speak(){
        cout << "Woof!" << endl;
    }
};

class Dog : public Animal{
    public:
    Dog (string name):Animal(name){}
    void speak(){
        cout << "Geo, Geo!" << endl;
    }
};
int main()
{
    
    Dog myDog("Tom");
    myDog.speak();
    
    Animal myAnimal("Shiva");
    myAnimal.speak();

    return 0;
}

File: 4_single_inheritance.py
class Animal :
    def __init__(self, name):
        self.name = name 
        
    def speak(self):
        print(f'{self.name} are making sound.\n')
        
    
class Dog(Animal):
    def speak(self):
        print(f'{self.name} are geo geo.\n')
        
        
        
dog = Dog('Kutta')
dog.speak()